src/lib/ampVariantAssignments.ts

```diff
@@ -0,0 +1,20 @@
+import express from 'express';
+
+export type AmpVariantAssignments = {
+    [key: string]: string;
+};
+
+export const getAmpVariantAssignments = (req: express.Request): AmpVariantAssignments => {
+    const { testData } = req.query;
+    const ampVariantAssignments: AmpVariantAssignments = {};
+
+    testData
+        ?.toString()
+        .split('!')
+        .forEach((testAndVariant: string) => {
+            const [test, variant] = testAndVariant.split('.');
+            ampVariantAssignments[test.toUpperCase()] = variant.toUpperCase();
+        });
+
+    return ampVariantAssignments;
+};
```

Comment:
Perhaps we should make the argument to this just the query string? That way it doesn't need to know anything about `express` and it would make it straight forward to write some tests for!

Reply:
Good shout @TomPretty! I'll do some refactoring now.

Reply:
Although, the way I've implemented it currently does mirror what's been done elsewhere in the codebase (https://github.com/guardian/support-dotcom-components/blob/88a5d5c7decabb99915090fbb7cadbee4cbd4800/src/lib/params.ts#L13).

I'd be inclined to stick with the current solution and refactor both at a later time.

@TomPretty @tomrf1 - thoughts?